lookout/style/typos/ranking.py

```diff
@@ -24,41 +25,33 @@ class CandidatesRanker(Model):
     VENDOR = "source{d}"
     DESCRIPTION = "Model that ranks candidates according to their probability to fix the typo."
     LICENSE = DEFAULT_LICENSE
-    DEFAULT_TRAIN_ROUNDS = 4000
-    DEFAULT_EARLY_STOPPING = 200
-    DEFAULT_BOOST_PARAM = {"max_depth": 6,
-                           "eta": 0.03,
-                           "min_child_weight": 2,
-                           "silent": 1,
-                           "objective": "binary:logistic",
-                           "nthread": 16,
-                           "subsample": 0.5,
-                           "colsample_bytree": 0.5,
-                           "alpha": 1,
-                           "eval_metric": ["error"]}
-
-    def __init__(self, **kwargs):
-        """Initialize a new instance of CandidatesRanker class."""
+
+    def __init__(self, config: Optional[Mapping[str, Any]] = None, **kwargs):
+        """
+        Initialize a new instance of CandidatesRanker class.
+
+        :param config: Ranking configuration, options:
+                       train_rounds: Number of training rounds.
+                       early_stopping: Early stopping parameter.
+                       boost_param: Boosting parameters.
+        :param kwargs: Extra keyword arguments which are consumed by Model.
+        """
         super().__init__(**kwargs)
-        self.train_rounds = self.DEFAULT_TRAIN_ROUNDS
-        self.early_stopping = self.DEFAULT_EARLY_STOPPING
-        self.boost_param = self.DEFAULT_BOOST_PARAM
+        self.set_config(config)
         self.bst = None  # type: xgb.Booster
 
-    def construct(self, boost_param: Optional[dict] = None,
-                  train_rounds: int = DEFAULT_TRAIN_ROUNDS,
-                  early_stopping: int = DEFAULT_EARLY_STOPPING) -> None:
+    def set_config(self, config: Optional[Mapping[str, Any]] = None) -> None:
```

Comment:
I do not think we should allow changing the config. Let's just inline this function into __init__.
Or rename it to `_set_config`,

For all `set_config` functions.

Reply:
I'm not shure this is so important. Both Ranker and Generator can be used several times with different configurations after init, and it won't break anything. In the same time loading other non-config data can time some time.
TLDR i want to use that feature while experimenting with configurations, so i think it has some value

Reply:
ok